sympy/core/evalf.py

```diff
@@ -1391,6 +1391,28 @@ def evalf(self, n=15, subs=None, maxn=100, chop=False, strict=False, quad=None,
             verbose=<bool>
                 Print debug information (default=False)
 
+        Note :
+        It is suggested to use expr.evalf(subs=...) over expr.subs(...).evalf()
+        or expr.evalf().subs(...) since expr.evalf(subs=...) avoids the loss of
+        significance caused due to naive substitution as the expression is run
+        through the evalf algorithm which takes into account, the various issues
+        that can cause loss of significance. Naive substitution can also include
+        substitution of only a few rather than all symbols in the expression.
+
+        Examples
+        ========
+
+        >>> import sympy
+        >>> x, y, z = sympy.symbols("x, y, z")
+        >>> (x+y-z).subs({x:1e100,y:1,z:1e100}) #case_1
+        0
+        >>> (x+y-z).evalf(subs={x: 1e100, y: 1, z: 1e100}) #case_2
+        1.00000000000000
+
+        in case_1, naive substitution evaluates 1e100 + 1 - 1e100 which looses 1 because of
```

Comment:
"In case ... which loses ..."

Reply:
Done. @sidhantnagpal  Pls review the new PR. I have applied all the changes you suggested